src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -82,8 +84,30 @@ public function autoEagerLoadRelationshipColumns()
     {
         $relationships = $this->getColumnsRelationships();
 
-        if (count($relationships)) {
-            $this->with($relationships);
+        foreach ($relationships as $relation) {
+            if (strpos($relation, '.') !== false) {
+                $relation_parts = explode('.', $relation);
+                $last_relation_part = $last_valid_relation_method = end($relation_parts);
+
+                array_reduce(array_splice($relation_parts, 0, count($relation_parts)), function ($obj, $method) use (&$last_valid_relation_method) {
```

Comment:
```suggestion
                array_reduce($relation_parts, function ($obj, $method) use (&$last_valid_relation_method) {
```
@pxpm can we do just this?

Because I think that `array_splice` isn't doing anything 😅